packages/kit/src/runtime/server/index.js

```diff
@@ -20,6 +20,10 @@ export async function respond(incoming, options, state = {}) {
 		};
 	}
 
+	if (!incoming.headers['user-agent']) {
```

Comment:
what's the reason for this check?

Reply:
Originally, I've put the code for the fix under the following condition...
https://github.com/sveltejs/kit/blob/92d33913e284e00fb20d882253ba5439eeafc407/packages/kit/src/runtime/server/page/load_node.js#L67-L68
...as that's the place where the problem happens. After Rich suggested the `index.js` file, I've put the code in there and added the additional condition, you've pointed at, cause the error happens only with server side requests.

The condition is there to make kind of a differentiation between client side and server side requests and is also there for performance reasons as client side header keys are all lowercase already.

A client could send a request without the 'user-agent' header, so the condition could potentially lead to the similar error in cases where requests without `user-agent` aren't automatically responded with for instance `405` status code upfront.

I'm not fond of modifying the very root file of the project as any addition done there will only make the overall performance of the app worse, but I'll do what you guys suggest in order to fix the problem.